kfp/kfp_support_lib/kfp_v1_workflow_support/Makefile

```diff
@@ -39,12 +39,15 @@ endif
 
 build-dist :: set-versions .defaults.build-dist
 
-publish:: .check-env
-	@# Help: Publish the wheel to testpypi
-	if [ -d "dist"]; then rm -r dist; fi
-	${PYTHON} -m pip install --upgrade build
-	${PYTHON} -m twine check dist/*
-	${PYTHON} -m twine upload --verbose --non-interactive dist/*
+publish::
```

Comment:
Why do we need it? We can publish the Python modules independently on defined KFP version

Reply:
The `build` rule in v1 library has this if statement, so if the kfp version is v2 it will not build the v1 lib. Thus, we added the same if statement for `publish`.
We can remove this check from both `build` and `publish`.

Reply:
Why are we building/publishing only one version of kfp at a time.  If the repo contains both, the top level `make publish` should publish both, no?

Reply:
I'm trying to shoot for a small set of comprehensive make targets that don't need configuration, so that for example, `make publish` publishes everything that is publishable from the repo.

Reply:
the Python libraries are published once. 
However, the kfp image depends on which library should be used, so it should be build and published twice.

Reply:
+1 @roytman The python libraries are used by both KFPV1 and KFPV2